yazi-core/src/tab/selected.rs

```diff
@@ -0,0 +1,251 @@
+use std::{collections::{BTreeSet, HashMap}, path::PathBuf};
+
+use yazi_shared::fs::Url;
+
+#[derive(Default)]
+pub struct Selected {
+	inner:   BTreeSet<Url>,
+	parents: HashMap<PathBuf, usize>,
+}
+
+impl Selected {
+	pub fn get_inner(&self)->BTreeSet<Url>{
+		return self.inner.clone()
+	}
+	pub fn new() -> Self { Selected { inner: BTreeSet::new(), parents: HashMap::new() } }
+
+	pub fn insert(&mut self, url: Url) -> bool { self.insert_many(&[&url]) }
+
+	pub fn insert_many(&mut self, urls: &[&Url]) -> bool {
+		if urls.is_empty() {
+			return false;
+		}
+		let url_buf = urls[0].to_path_buf();
+
+		let mut current_path = url_buf.clone();
+		while let Some(parent) = current_path.parent() {
+			if self.inner.contains(&Url::from(parent)) {
+				return false;
+			}
+			current_path = parent.to_path_buf();
+		}
+
+		if self.parents.contains_key(&url_buf) {
+			return false;
+		}
+
+		let mut current_path = url_buf.clone();
+		let len_of_urls = urls.len();
+		loop {
+			current_path = match current_path.parent() {
+				Some(parent) => parent.to_path_buf(),
+				None => break,
+			};
+			let counter = self.parents.entry(current_path.clone()).or_insert(0);
+			*counter += len_of_urls;
```

Comment:
Additionally, in this line, I changed from adding 1 to adding the length of `urls`.